btree/btree.c

```diff
@@ -5906,16 +6013,252 @@ static void btree_ut_num_threads_num_trees_kv_oper(uint32_t thread_count,
+	m0_bcount_t            vsize = sizeof value;
+	void                  *k_ptr = &key;
+	void                  *v_ptr = &value;
+	struct m0_btree_rec    rec   = {
+				     .r_key.k_data = M0_BUFVEC_INIT_BUF(&k_ptr,
+									&ksize),
+				     .r_val        = M0_BUFVEC_INIT_BUF(&v_ptr,
+									&vsize),
+				     .r_flags      = 0,
+				     };
+	struct cb_data         data  = {
+					.key         = &rec.r_key,
+					.value       = &rec.r_val,
+					.check_value = false,
+					.flags       = 0,
+				       };
+	struct m0_btree_cb     ut_cb   = {
+					  .c_act       = btree_kv_put_cb,
+					  .c_datum     = &data,
+					 };
+	int32_t                loop_count;
+	struct m0_btree_op     kv_op     = {};
+	void                  *temp_node;
+	struct m0_btree_type   btree_type  = {.tt_id = M0_BT_UT_KV_OPS,
+					      .ksize = sizeof(key),
+					      .vsize = sizeof(value),
+					     };
+	const struct node_type *nt         = &fixed_format;
+	struct m0_be_tx        *tx         = NULL;
+	int                     rc;
+
+	random_r(&ti->ti_random_buf, &loop_count);
+	loop_count %= (MAX_TREE_LOOPS - MIN_TREE_LOOPS);
+	loop_count += MIN_TREE_LOOPS;
+
+	while(!thread_start)
+		;
+
+	/** Create temp node space and use it as root node for btree */
+	temp_node = m0_alloc_aligned((1024 + sizeof(struct nd)), 10);
+
+	while (loop_count--) {
+		struct m0_btree_op  b_op        = {};
+		struct m0_btree    *tree;
+		int32_t             rec_count;
+		uint32_t            i;
+
+		/**
+		 * 1) Create a tree
+		 * 2) Add a few random count of records in the tree.
+		 * 3) Close the tree
+		 * 4) Open the tree
+		 * 5) Confirm the records are present in the tree.
+		 * 6) Close the tree
+		 * 4) Open the tree
+		 * 5) Delete all the records from the tree.
+		 * 6) Close the tree
+		 * 7) Destroy the tree
+		 */
+
+		rc = M0_BTREE_OP_SYNC_WITH_RC(&b_op.bo_op,
+					      m0_btree_create(temp_node, 1024,
+							      &btree_type, nt,
+							      &b_op, tx),
+					      &b_op.bo_sm_group,
+					      &b_op.bo_op_exec);
+		M0_ASSERT(rc == 0);
+
+		tree = b_op.bo_arbor;
+
+		random_r(&ti->ti_random_buf, &rec_count);
+		rec_count %= MAX_RECS_FOR_TREE_TEST;
+		rec_count = rec_count ? : (MAX_RECS_FOR_TREE_TEST / 2);
+
+		for (i = 1; i <= rec_count; i++) {
+			value = key = i;
+
+			rc = M0_BTREE_OP_SYNC_WITH_RC(&kv_op.bo_op,
+						      m0_btree_put(tree, &rec,
+								   &ut_cb, 0,
+								   &kv_op, tx),
+						      &kv_op.bo_sm_group,
+						      &kv_op.bo_op_exec);
+			M0_ASSERT(data.flags == M0_BSC_SUCCESS && rc == 0);
+		}
+
+		m0_btree_close(tree);
+
+		rc = M0_BTREE_OP_SYNC_WITH_RC(&kv_op.bo_op,
+					      m0_btree_open(temp_node,
+							    1024, &tree,
+							    &kv_op),
+					      &kv_op.bo_sm_group,
+					      &kv_op.bo_op_exec);
+		M0_ASSERT(rc == 0);
+
+		for (i = 1; i <= rec_count; i++) {
+			value = key = i;
+
+			rc = M0_BTREE_OP_SYNC_WITH_RC(&kv_op.bo_op,
+						      m0_btree_get(tree,
+								   &rec.r_key,
+								   &ut_cb,
```

Comment:
you are using btree_kv_put_cb for callback you might want to use btree_kv_get_cb